AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3148,6 +3148,164 @@ function Debug-AzStorageAccountADObject
+        {
+            if ($kerberosTicketEncryptionServerNoDash.Contains($type)) 
+            {
+                $found = $true
+                break
+            }
+        }
+
+        if (!$found) 
+        {
+            Write-Error -Message "The server side and the client side do not have a Kerberos Ticket Encryption type in common." -ErrorAction Stop
+        }
+
+    }
+}
+
+function Debug-ChannelEncryption
+{
+    [CmdletBinding()]
+
+    param (
+        [Parameter(Mandatory=$True, Position=0, HelpMessage="Storage account name")]
+        [string]$StorageAccountName,
+
+        [Parameter(Mandatory=$True, Position=1, HelpMessage="Resource group name")]
+        [string]$ResourceGroupName
+    )
+
+    process
+    {
+
+        $storageAccount = Validate-StorageAccount -ResourceGroupName $ResourceGroupName -StorageAccountName $StorageAccountName -ErrorAction Stop
+
+        $protocolSettings = (Get-AzStorageFileServiceProperty -StorageAccount $storageAccount -ErrorAction Stop).ProtocolSettings.Smb
+
+        $channelEncryptionsClient = (Get-SmbServerConfiguration).EncryptionCiphers.replace("_", "-")
+
+        $channelEncryptionsServer = $protocolSettings.ChannelEncryption
+        if ($null -eq $channelEncryptionsServer)
+        {
+            # if null, all types are supported for the storage account
+            $channelEncryptionsServer = "AES-128-CCM", "AES-128-GCM", "AES-256-GCM"
+        }
+        $channelEncryptionsServerWithComma = [String]::Join(", ", $channelEncryptionsServer)
+
+        Write-Host "Channel Encryption Supported on the Client Side: $channelEncryptionsClient"
+        Write-Host "Channel Encryption Supported on the Server Side: $channelEncryptionsServerWithComma"
+
+        $found = $false
+        foreach($type in $channelEncryptionsServer)
+        {
+            if($channelEncryptionsClient.Contains($type))
+            {
+                $found = $true
+                break    
+            }
+        }
+
+        if(!$found)
+        {
+            Write-Error -Message "The server side and the client side do not have a Channel Encryption type in common." -ErrorAction Stop
+        }
+        
+    }
+}
+
+function Debug-DomainLineOfSight
+{
+    [CmdletBinding()]
+
+    param (
+        [Parameter(Mandatory=$True, Position=0, HelpMessage="Storage account name")]
+        [string]$StorageAccountName,
+
+        [Parameter(Mandatory=$True, Position=1, HelpMessage="Resource group name")]
+        [string]$ResourceGroupName
+    )
+
+    process
+    {
+        $storageAccount = Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName
+        $fullyQualifiedDomainName = $storageAccount.AzureFilesIdentityBasedAuth.ActiveDirectoryProperties.DomainName
+        Write-Host "Fully Qualified Domain Name: $fullyQualifiedDomainName"
+        $checkResult = nltest /dsgetdc:$fullyQualifiedDomainName | Out-String
+
+        if([string]::IsNullOrEmpty($checkResult))

```

Comment:
There will be an error message if nltest fails, so this check is incorrect. You need to parse the output to validate the result.

Reply:
That was my original approach, but I always got NULL as a result when I entered a non-reachable fqdn. I will continue to look into it.

Reply:
$checkResult = nltest /dsgetdc:$fullyQualifiedDomainName | Out-String

After the testing during our discussion, it's proven that nltest's failure message doens't work as the error messages thrown by all other cmdlets. If it fails, we receive null for the result, and not a string with the error message.